pkg/api/addtask.go

```diff
@@ -0,0 +1,76 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/sudodju/go_final_project/pkg/db"
+)
+
+// Функция проверки правильности формата даты
+func checkDate(task *db.Task) (time.Time, error) {
+	var t time.Time
+	var err error
+	if t, err = time.Parse(dateFormat, task.Date); err != nil {
+		return t, fmt.Errorf("Неверно указан формат даты: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return t, fmt.Errorf("Неверно указан формат даты: %w", err)
```